datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -740,6 +804,47 @@ pub async fn from_substrait_rex(
                 "Cast experssion without output type is not allowed".to_string(),
             )),
         },
+        Some(RexType::WindowFunction(window)) => {
+            let fun = match extensions.get(&window.function_reference) {
+                Some(function_name) => Ok(find_df_window_func(function_name)),
```

Comment:
Should we return an error if `find_df_window_func` returns `None` here? We `unwrap` this later on currently, which could fail?